src/hooks/RestrictedTransfers.sol

```diff
@@ -67,13 +70,13 @@ contract RestrictedTransfers is Auth, IRestrictedTransfers, IHook {
         view
         returns (bool)
     {
-        if (uint128(hookData.from).getBit(FREEZE_BIT) == true && !root.endorsed(from)) {
+        if (uint128(hookData.from).getBit(FREEZE_BIT) == true && !root.endorsed(from) && from != ESCROW_IDENTIFIER) {
             // Source is frozen and not endorsed
             return false;
         }
 
-        if (root.endorsed(to) || to == address(0)) {
-            // Destination is endorsed and source was already checked, so the transfer is allowed
+        if (root.endorsed(to) || to == address(0) || to == ESCROW_IDENTIFIER) {
```

Comment:
NOTE: These changes were necessary because previously the global escrow contract was endorsed by root during deployment which we cannot do anymore as escrows are deployed dynamically by the corresponding factory.